baseplate/lib/metrics.py

```diff
@@ -112,17 +112,25 @@ def flush(self) -> None:
 class RawTransport(Transport):
     """A transport which sends messages on a socket."""
 
-    def __init__(self, endpoint: config.EndpointConfiguration):
+    def __init__(
+        self, endpoint: config.EndpointConfiguration, swallow_network_errors: bool = False,
+    ):
+        self.swallow_network_errors = swallow_network_errors
         self.socket = socket.socket(endpoint.family, socket.SOCK_DGRAM)
         self.socket.connect(endpoint.address)
 
     def send(self, serialized_metric: bytes) -> None:
         try:
-            self.socket.sendall(serialized_metric)
-        except socket.error as exc:
-            if exc.errno == errno.EMSGSIZE:
-                raise MessageTooBigTransportError(len(serialized_metric))
-            raise TransportError(exc)
+            try:
+                self.socket.sendall(serialized_metric)
+            except socket.error as exc:
+                if exc.errno == errno.EMSGSIZE:
+                    raise MessageTooBigTransportError(len(serialized_metric))
+                raise TransportError(exc)
+        except Exception:
+            if not self.swallow_network_errors:
```

Comment:
the configurable is "swallow _network_ errors" but we've added a new try/except here that's far broader than network errors. was there a non-socket.error exception you were running into in practice that made you want the broader catch? or would it work to add this `if` around the `raise TransportError` above? i'm just worried about swallowing true programming errors here rather than purely operational stuff

Reply:
good point. will do.

Reply:
Done.